core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -793,8 +792,14 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
                 }
             }
         }
-
-        Map<String, Float> newFieldsBoosts = QueryParserHelper.resolveMappingFields(context, fieldsBoosts);
+        Map<String, Float> newFieldsBoosts;
+        if (fieldsBoosts.isEmpty()) {
+            // no fields provided, defaults to index.query.default_field
+            List<String> defaultFields = context.defaultFields();
+            newFieldsBoosts = QueryParserHelper.resolveMappingFields(context, QueryParserHelper.parseFieldsAndWeights(defaultFields));
```

Comment:
Does this need to automatically set `lenient: true` here also?

Reply:
Good catch. I pushed another commit to set `lenient: true` if the default field is `*`.